deepfence_bootstrapper/main.go

```diff
@@ -124,19 +156,51 @@ func main() {
 			log.Warn().Msg("Failed to authenticate. Retrying...")
 			time.Sleep(authCheckPeriod)
 		} else {
+			supervisor.StopAllProcesses()
 			log.Panic().Msgf("Fatal: %v", err)
 		}
 	}
+
+	//If cloud agent, Get the node id from the cloudscanner
+	cloudNodeID := ""
+	if enableCloudNode {
+		for i := 0; i < 10; i++ {
+			log.Info().Msgf("Getting the GetCloudNodeID")
+			cloudNodeID, err = router.GetCloudNodeID()
+			if err != nil {
+				log.Error().Msgf("Error getting CloudNode ID, RetryCount: %d, error: %v", i, err)
+			} else if len(cloudNodeID) == 0 {
+				err = fmt.Errorf("Empty CloudNode ID received")
+				log.Error().Msgf("Error: %s, RetryCount: %d", err.Error(), i)
+			} else {
+				err = nil
+				hostname = cloudNodeID
+				log.Info().Msgf("Cloud mode, setting hostname as: %s", cloudNodeID)
+				break
+			}
+			time.Sleep(10 * time.Second)
+		}
+		if err != nil {
+			//Fatal Failed to get the CloudNodeID
+			supervisor.StopAllProcesses()
+			log.Fatal().Msgf("Failed to get CloudNode, error: %s", err.Error())
+		}
+	}
+
 	for {
 		if enableClusterDiscovery {
 			k8sClusterID, _, _, _, _ := dfUtils.GetKubernetesDetails()
-			err = consoleClient.StartControlsWatching(k8sClusterID, true, Version)
 			log.Info().Msgf("cluster agent mode: %s", k8sClusterID)
+			err = consoleClient.StartControlsWatching(k8sClusterID, true, Version, ctl.K8S)
+		} else if enableCloudNode {
+			log.Info().Msgf("cloud agent mode: %s", hostname)
+			err = consoleClient.StartControlsWatching(hostname, false, Version, ctl.CLOUD_AGENT)
 		} else {
-			err = consoleClient.StartControlsWatching(hostname, false, Version)
 			log.Info().Msgf("regular agent mode: %s", hostname)
+			err = consoleClient.StartControlsWatching(hostname, false, Version, ctl.HOST)
 		}
```

Comment:
Shall we clean that up & have 1 call to `StartControlsWatching` instead pf 3

Reply:
Done.